component/s3storage/client.go

```diff
@@ -155,7 +148,30 @@ func (cl *Client) Configure(cfg Config) error {
 		log.Warn("Client::Configure : Bucket defaulted to first listed bucket: %s", bucketList[0])
 	}
 
-	return err
+	// Use list objects validate the region and bucket access
+	_, _, err = cl.List("/", nil, 1)
+	if err != nil {
+		log.Err("Client::Configure : listing objects failed. Here's why: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func getRegionFromEndpoint(endpoint string) (string, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return "", err
+	}
```

Comment:
We should explicitly check for an empty endpoint, since that won't return an error here.

Reply:
Good point.